Mathlib/MeasureTheory/Integral/Lebesgue/SigmaFinite.lean

```diff
@@ -0,0 +1,201 @@
+    (ε0 : ε ≠ 0) : ∃ g : α → ℝ≥0, (∀ x, 0 < g x) ∧ Measurable g ∧ ∫⁻ x, g x ∂μ < ε := by
+  /- Let `s` be a covering of `α` by pairwise disjoint measurable sets of finite measure. Let
+    `δ : ℕ → ℝ≥0` be a positive function such that `∑' i, μ (s i) * δ i < ε`. Then the function that
+     is equal to `δ n` on `s n` is a positive function with integral less than `ε`. -/
+  set s : ℕ → Set α := disjointed (spanningSets μ)
+  have : ∀ n, μ (s n) < ∞ := fun n =>
+    (measure_mono <| disjointed_subset _ _).trans_lt (measure_spanningSets_lt_top μ n)
+  obtain ⟨δ, δpos, δsum⟩ : ∃ δ : ℕ → ℝ≥0, (∀ i, 0 < δ i) ∧ (∑' i, μ (s i) * δ i) < ε :=
+    ENNReal.exists_pos_tsum_mul_lt_of_countable ε0 _ fun n => (this n).ne
+  set N : α → ℕ := spanningSetsIndex μ
+  have hN_meas : Measurable N := measurable_spanningSetsIndex μ
+  have hNs : ∀ n, N ⁻¹' {n} = s n := preimage_spanningSetsIndex_singleton μ
+  refine ⟨δ ∘ N, fun x => δpos _, measurable_from_nat.comp hN_meas, ?_⟩
+  erw [lintegral_comp measurable_from_nat.coe_nnreal_ennreal hN_meas]
+  simpa [N, hNs, lintegral_countable', measurable_spanningSetsIndex, mul_comm] using δsum
+#align measure_theory.exists_pos_lintegral_lt_of_sigma_finite MeasureTheory.exists_pos_lintegral_lt_of_sigmaFinite
+
+section Trim
+
+variable {m0 : MeasurableSpace α} {μ : Measure α}
+
+theorem univ_le_of_forall_fin_meas_le (hm : m ≤ m0) [SigmaFinite (μ.trim hm)]
+    (C : ℝ≥0∞) {f : Set α → ℝ≥0∞} (hf : ∀ s, MeasurableSet[m] s → μ s ≠ ∞ → f s ≤ C)
+    (h_F_lim : ∀ S : ℕ → Set α, (∀ n, MeasurableSet[m] (S n)) → Monotone S →
+      f (⋃ n, S n) ≤ ⨆ n, f (S n)) :
+    f univ ≤ C := by
+  let S := @spanningSets _ m (μ.trim hm) _
+  have hS_mono : Monotone S := @monotone_spanningSets _ m (μ.trim hm) _
+  have hS_meas : ∀ n, MeasurableSet[m] (S n) := @measurable_spanningSets _ m (μ.trim hm) _
+  rw [← @iUnion_spanningSets _ m (μ.trim hm)]
+  refine (h_F_lim S hS_meas hS_mono).trans ?_
+  refine iSup_le fun n => hf (S n) (hS_meas n) ?_
+  exact ((le_trim hm).trans_lt (@measure_spanningSets_lt_top _ m (μ.trim hm) _ n)).ne
+#align measure_theory.univ_le_of_forall_fin_meas_le MeasureTheory.univ_le_of_forall_fin_meas_le
+
+/-- If the Lebesgue integral of a function is bounded by some constant on all sets with finite
+measure in a sub-σ-algebra and the measure is σ-finite on that sub-σ-algebra, then the integral
+over the whole space is bounded by that same constant. Version for a measurable function.
+See `lintegral_le_of_forall_fin_meas_le'` for the more general `AEMeasurable` version. -/
+theorem lintegral_le_of_forall_fin_meas_le_of_measurable (hm : m ≤ m0) [SigmaFinite (μ.trim hm)]
+    (C : ℝ≥0∞) {f : α → ℝ≥0∞} (hf_meas : Measurable f)
+    (hf : ∀ s, MeasurableSet[m] s → μ s ≠ ∞ → ∫⁻ x in s, f x ∂μ ≤ C) : ∫⁻ x, f x ∂μ ≤ C := by
+  have : ∫⁻ x in univ, f x ∂μ = ∫⁻ x, f x ∂μ := by simp only [Measure.restrict_univ]
+  rw [← this]
+  refine univ_le_of_forall_fin_meas_le hm C hf fun S hS_meas hS_mono => ?_
```

Comment:
The rest of the proof is about `∫⁻ x in ⋃ n, S n, f x = ⨆ n, ∫⁻ x in S n, f x`. I'm going to move it to a separate lemma, then this file becomes much shorter, so I'm going to look for another way to split the original file.